server/coderadar-webapp/src/main/java/org/wickedsource/coderadar/security/domain/UserCredentialsResource.java

```diff
@@ -0,0 +1,34 @@
+package org.wickedsource.coderadar.security.domain;
+
+import org.springframework.hateoas.ResourceSupport;
+
+public class UserCredentialsResource extends ResourceSupport {
+
+    private String username;
```

Comment:
add @NotNull to username and password field and add @Valid to parameter in register() function of RegistrationController for automatic validation.